src/main/java/com/gmail/inverseconduit/bot/DefaultBot.java

```diff
@@ -52,6 +53,13 @@ private void processMessageQueue() {
     }
 
     private void processMessage(final ChatMessage chatMessage) {
+    	for (final CommandHandle listener : listeners){
+    		final String response = listener.execute(chatMessage);
+    		if (response != null){
```

Comment:
Oh yes... you're right. filter out the null values then:

    listeners.stream().map(l -> l.execute(chatMessage)).filter(l -> null == l).forEach(chatInterface.sendMessage(...));